src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,645 @@
+
+            headerNodes[3].NameNode.Text.Should().Be("ContentLength");
+            headerNodes[3].SeparatorNode.Text.Should().Be(":");
+            headerNodes[3].ValueNode.Text.Should().Be("7060");
+
+            headerNodes[4].NameNode.Text.Should().Be("Cookie");
+            headerNodes[4].SeparatorNode.Text.Should().Be(":");
+            headerNodes[4].ValueNode.Text.Should().Be("expor=;HSD=Ak_1ZasdqwASDASD;SSID=SASASSDFsdfsdf213123;APISID=WRQWRQWRQWRcc123123;");
+        }
+    }
+
+    public class Body
+    {
+        [Fact]
+        public void header_with_body_is_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+        POST https://example.com/comments HTTP/1.1
+        Content-Type: application/xml
+        Authorization: token xxx
+
+        <request>
+            <name>sample</name>
+            <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+        </request>
+        """);
+
+            using var _ = new AssertionScope();
+
+            var requestNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.HeaderNodes.Count.Should().Be(2);
+            requestNode.BodySeparatorNode.TextWithTrivia.Should().Be("\r\n");
+            requestNode.BodyNode.Text.Should().Be(
+                """
+        <request>
+            <name>sample</name>
+            <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+        </request>
+        """);
+        }
+
+        [Fact]
+        public void body_without_header_header_is_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+    POST https://example.com/comments HTTP/1.1
+
+    <request>
+        <name>sample</name>
+        <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+    </request>
+    """);
+
+            using var _ = new AssertionScope();
+
+            var requestNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.HeaderNodes.Count.Should().Be(0);
+        }
+
+        //TODO: Is it an error to include a body with no headers?
+        [Fact]
+        public void body_without_header_body_is_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+        POST https://example.com/comments HTTP/1.1
+
+        <request>
+            <name>sample</name>
+            <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+        </request>
+        """);
+
+            using var _ = new AssertionScope();
+
+            var requestNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.BodyNode.Text.Should().Be(
+                """
+        <request>
+            <name>sample</name>
+            <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+        </request>
+        """);
```

Comment:
minor nit: Same comment as above about indentation for the multi-line string literals being off...